save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -70,123 +76,151 @@ class WarnPlugin(
+                    it.toString().makeThePathRelativeToTestRoot()
+                }
+            }
+
+        val execCmd = "${generalConfig.execCmd} ${warnPluginConfig.execFlags} $fileNamesForExecCmd"
+
         val executionResult = try {
-            pb.exec(execCmd, null)
+            pb.exec("cd ${testConfig.getRootConfig().location.parent} && " + execCmd, null)
         } catch (ex: ProcessExecutionException) {
-            return listOf(TestResult(
-                paths,
-                Fail(ex.describe(), ex.describe()),
-                DebugInfo(null, ex.message, null)
-            ))
+            return sequenceOf(
+                TestResult(
+                    paths,
+                    Fail(ex.describe(), ex.describe()),
+                    DebugInfo(null, ex.message, null)
+                )
+            )
         }
         val stdout = executionResult.stdout
         val stderr = executionResult.stderr
 
         val actualWarningsMap = executionResult.stdout.mapNotNull {
             with(warnPluginConfig) {
-                val line = it.getLineNumber(warningsOutputPattern!!, lineCaptureGroupOut, linePlaceholder!!, null)
-                it.extractWarning(warningsOutputPattern, fileNameCaptureGroupOut!!, line, columnCaptureGroupOut, messageCaptureGroupOut!!)
+                val line = it.getLineNumber(actualWarningsPattern!!, lineCaptureGroupOut, linePlaceholder!!, null)
+                it.extractWarning(
+                    actualWarningsPattern,
+                    fileNameCaptureGroupOut!!,
+                    line,
+                    columnCaptureGroupOut,
+                    messageCaptureGroupOut!!
+                )
             }
         }
-            .groupBy { if (it.line != null && it.column != null) it.line to it.column else null }
+            .groupBy { it.fileName }
             .mapValues { (_, warning) -> warning.sortedBy { it.message } }
 
         return paths.map { path ->
             TestResult(
                 listOf(path),
                 checkResults(
-                    expectedWarnings.filter { it.value.any { warning -> warning.fileName == path.name } },
-                    actualWarningsMap.filter { it.value.any { warning -> warning.fileName == path.name } },
+                    expectedWarnings[path.name] ?: listOf(),
+                    actualWarningsMap[path.name] ?: listOf(),
                     warnPluginConfig
                 ),
                 DebugInfo(
                     stdout.filter { it.contains(path.name) }.joinToString("\n"),
                     stderr.filter { it.contains(path.name) }.joinToString("\n"),
-                    null)
+                    null
+                )
             )
-        }
+        }.asSequence()
     }
 
-    /**
-     * @param path
-     * @param warningsInputPattern
-     * @return name of the temporary file
-     */
-    internal fun createTestFile(path: Path, warningsInputPattern: Regex): String {
```

Comment:
we don't want to create a new test file in WARN plugin, wtf, it is not needed, we should not run with incorrect line numbers